test/cpp/qps/driver.cc

```diff
@@ -492,6 +492,10 @@ std::unique_ptr<ScenarioResult> RunScenario(
   Histogram merged_latencies;
   std::unordered_map<int, int64_t> merged_statuses;
 
+  // For the case where clients leads the test, it's going to finish
+  // clients first and wait until it's completely done for generaous termination.
+  bool client_finish_first = (client_config.rpc_type() != STREAMING_FROM_SERVER);
+
```

Comment:
Can you document why this is the case? I'm not sure that I understand the difference here (it's been a while since I saw the code)